### PR TITLE
POC support update with expressions.

### PR DESCRIPTION
**Pull request by WaVEV**
This is a quick approach to support expression in updates. I have to say that the update pipeline is not expressive as it's needed

### CODE REVIEW

File: django_mongodb/compiler.py [hunk 1]
```diff
@@ -591,42 +603,44 @@ def execute_sql(self, result_type):
...
+        try:
+            criteria = self.build_query().mongo_query
+        except EmptyResultSet:
+            return 0
         is_empty = not bool(values)
-        rows = 0 if is_empty else self.update(values)
+        if is_empty:
+            rows = 0
+        else:
+            base_pipeline = [
```
Comment (WaVEV) [inline django_mongodb/compiler.py:617]: This part is the only thing that need focus. Here I just create the **$merge** pipeline and the **affected rows** pipeline with a single transaction. I really don't know if this is the way to go.

Comment (Jibola) [reply django_mongodb/compiler.py:617]: This is the right way to go about it. having a transaction ensures that state isn't changed between the lookup and the update.

File: django_mongodb/compiler.py [hunk 2]
```diff
@@ -591,42 +603,44 @@ def execute_sql(self, result_type):
...
+                    "$merge": {
+                        "into": self.collection_name,
+                        "whenMatched": "replace",
+                        "whenNotMatched": "discard",
+                    }
+                },
+            ]
+            with self.connection.connection.start_session() as session, session.start_transaction():
+                result = next(self.collection.aggregate(count_pipeline), {"count": 0})
+                self.collection.aggregate(pipeline)
```
Comment (Jibola) [inline django_mongodb/compiler.py:634]: what has been the output when you merge:
`[*base_pipeline, {$merge..}, {$count...}]` Does this not work whatsoever?

**Review by Jibola — COMMENTED** [review]
Overall, agree with the approach. I'm *hoping* there is another way to combine the count and merge _after_ the merge has occurred, since we ultimately just want the number of documents that were changed.

File: django_mongodb/compiler.py (hunk 1 above)
Comment (WaVEV) [reply django_mongodb/compiler.py:617]: I think we can use update many directly as Shane mentioned. If I've re-read the docs  and what the docs support are this three stages.

Comment (WaVEV) [reply django_mongodb/compiler.py]: Yeap, It works 🚀. So we are able to do the update without making two queries (one for count and the other for the update)

**Comment by WaVEV** [comment]
The final implementation is in #112

**Pull request closed without merging**